src/quota_aggregator_impl.cc

```diff
@@ -137,6 +142,18 @@ ::google::protobuf::util::Status QuotaAggregatorImpl::Quota(
   // Aggregate tokens if the cached response is positive
   if (lookup.value()->is_positive_response()) {
```

Comment:
So the logic in Quota() is

lookup_cache, if not_found,  {
   create a cache item
  AddRemovedItem() to send call to server
   return positive
}

Check last_refresh_time, if too old {
    AddRemoveItem() to send refresh time.
} else {
     if positive {
           aggregate
    }
}
return cached_response;